cpp/src/io/orc/reader_impl.hpp

```diff
@@ -71,18 +66,50 @@ class reader::impl {
                            std::vector<std::vector<size_type>> const& stripes);
 
  private:
+  /**
+   * @brief Perform the necessary data preprocessing for parsing file later on.
```

Comment:
we perform decoding in this function, I don't see how it's preparing for further processing. The only work we do after this function is to create columns out of the decoded buffers.

Reply:
Yes this function should be everything here. I was planning for the chunked reader thus the docs is ahead of time. Let me change it back.